contracts/vaults/BaseVaults/RibbonThetaVault.sol

```diff
@@ -322,6 +335,22 @@ contract RibbonThetaVault is RibbonVault, RibbonThetaVaultStorage {
         );
     }
 
+    /**
+     * @notice Stakes a users vault shares
+     * @param numShares is the number of shares to stake
+     */
+    function stake(uint256 numShares) external nonReentrant {
+        require(stakingRewards != address(0)); // Removed revert msgs due to contract size limit
+        require(numShares > 0);
+        uint256 heldByAccount = balanceOf(msg.sender);
+        if (heldByAccount < numShares) {
+            _redeem(numShares.sub(heldByAccount), false);
+        }
+        _transfer(msg.sender, address(this), numShares);
+        _approve(address(this), stakingRewards, numShares);
```

Comment:
Instead of doing an intermediate `_transfer` step to `address(this)`, we could approve on behalf of the user, eg:

```solidity
_approve(msg.sender, stakingRewards, numShares);
IStakingRewards(stakingRewards).stakeFor(numShares, msg.sender);

// for security after stakeFor, we can verify that the stakingRewards behaves as it should be
require(balanceOf(stakingRewards).sub(startStakingRewardsBalance) == numShares, "Wrong amount staked");
```